x/escrow/codec.proto

```diff
@@ -16,9 +18,11 @@ message Escrow {
   bytes sender = 1;
   bytes arbiter = 2;
   bytes recipient = 3;
-  // if unreleased before timeout, will return to sender
-  // timeout stored here is absolute block height
-  int64 timeout = 5;
+
+  // If unreleased before timeout, escrow will return to sender.
+  // Timeout represents wall clock time as read from the block header.
+  google.protobuf.Timestamp timeout = 5 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
```

Comment:
q: Do we want to replace block timeout by time? Both would be valid options IMHO

Reply:
:man_shrugging: Who can answer this question?

Reply:
In case of Ethereum, we have no reliable wall time available. With this change, all block height based timeout in client code breaks, which leads to the issue that we need to support different kind of timeouts of different chains. Given that we have no experience with setting those timeouts and no overview of all the timeout related attack vectors, I personally think it is too early to introduce the concept of wall clock timeouts.

Reply:
Wall time is much more reliable for the bns than block height is.

As block times can vary between 1s and 5m in our setup. But the header timestamp is agreed within small limits by 2/3 of the validators.

For ethereum, please continue to use block height.

Reply:
For iov-core, you could expose a time on the Transaction (the user's intention) and then let the chain convert it to a predicted block height if needed... Maybe a bit trickier to parse though.